antarest/study/business/binding_constraint_management.py

```diff
@@ -111,9 +112,19 @@ def generate_id(self) -> str:
         return self.data.generate_id()
 
 
-class UpdateBindingConstProps(BaseModel):
-    key: str
-    value: Any
+class BindingConstraintEditionModel(BaseModel, metaclass=AllOptionalMetaclass):
+    group: str
+    enabled: bool
+    time_step: BindingConstraintFrequency
+    operator: BindingConstraintOperator
+    filter_year_by_year: str
+    filter_synthesis: str
+    comments: str
+    coeffs: Dict[str, List[float]]
```

Comment:
Same thing as the comment below, the commands create and update have the field `coeffs`.

Reply:
The input you pass to commands is not the same as the model represented by Pydantic, you can have a representation of a given model that you want to validate, and construct the data you want to persist differently.

Pydantic models are primarily used for data validation and serialization, there might be scenarios where the structure of data coming from or going to the database needs to be transformed or adapted for various reasons.

Given that the `coeffs` field (`Dict[str, List[float]]`)  you pass to commands is not like `constraints` (`List[ConstraintTermDTO]`) that comes and goes to the client. So you have at least to change it from coeffs to constraints (only in the model, of course you keep it coeffs for the commands)

Reply:
And the reason why I suggested to remove `constraints` from the Input model, is cause it represents the constraint terms. And the constraints terms are handled separately to the constraint properties. (they are never required both at the same time) so we can easily dissociate them.

But you're right it's maybe a bad idea, we never know what it could breaks